javascript/code-challenges/graph/graph.js

```diff
@@ -0,0 +1,41 @@
+class Graph {
+  constructor() {
+    this.nodes = new Map();
+  }
+
+  addNode(value) {
+    if (!this.nodes.has(value)) {
+      this.nodes.set(value, []);
+    }
+    return value;
+  }
+
+  addEdge(node1, node2, weight = 1) {
+    if (!this.nodes.has(node1) || !this.nodes.has(node2)) {
+      throw new Error('Both nodes should already be in the graph');
+    }
+    const neighbors1 = this.nodes.get(node1);
+    const neighbors2 = this.nodes.get(node2);
+
+    neighbors1.push({ node: node2, weight });
+    neighbors2.push({ node: node1, weight });
```

Comment:
This would make an undirected graph. Just be aware that in a directed graph, only node1 gets an edge added to its neighbors list.